[Build] Don't compile PoW mining RPC commands by default

Fuzzbawls opened the pull request:
This adds a configure flag to skip compiling the PoW-related mining RPC
commands, which aren't ever used outside of the initial PoW phase.

The `generate` command is retained for regtest compatibility.

random-zebra reviewed (approved):
ACK 624fdff09d7da6138d32b98fe4c32d15728aa157.

Maybe it's worth to document this new configure flag `--enable-mining-rpc` in the `build-*` notes, or somewhere else (at least in the release notes).

Fuzzbawls commented:
rebased and added a note in the release notes template. Not going to add anything to the `build-*.md` files as this is an edge use case and `configure --help` provides all the documentation necessary.

random-zebra reviewed (approved):
re-utACK 1a5d90335eea7234fd24b1bc477f8b3fca1fdb0b

furszy reviewed (approved):
utACK 1a5d903 and merging
